src/main/java/jarvis/Formatter.java

```diff
@@ -0,0 +1,193 @@
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void eventCommand(String[] userCommand) throws JarvisException {
+        int indexOfAt = indexOf(userCommand, "/at");
+        int numOfArgs = userCommand.length;
+        boolean isValidIndex = indexOfAt > 1;
+        boolean hasSufficientArgs = numOfArgs >= 4;
+
+        if (isValidIndex && hasSufficientArgs) {
+            String eventDescription = parseUserInput(userCommand, 1, indexOfAt);
+            String eventDate = parseUserInput(userCommand, indexOfAt + 1, numOfArgs);
+            Event newEvent = new Event(eventDescription, eventDate);
+            UserList.insertTask(newEvent);
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void deleteCommand(String[] userCommand) throws JarvisException {
+        try {
+            Integer taskIndex = Integer.parseInt(userCommand[1]);
+            UserList.removeTask(taskIndex - 1);
+        } catch (NumberFormatException e){
+            throw new JarvisException();
+        } catch (IndexOutOfBoundsException e) {
+            throw new JarvisException();
+        }
+    }
+
+    public static void inputHandler(Scanner in) {
+        String inputLine = in.nextLine();
+        String[] userCommand = inputLine.split(" ");
+
+        switch (userCommand[0]) {
+        case "bye": //exit command
+            DisplayMessages.savingData();
+            UserList.saveData();
+            DisplayMessages.closingMessage();
+            System.exit(0);
+            break;
+
+        case "list":
+            UserList.printList();
+            break;
+
+        case "mark":
+            try {
+                markCommand(userCommand);
+            } catch (JarvisException e) {
+                DisplayMessages.outOfBounds();
+            }
+            break;
+
+        case "unmark":
+            try {
+                unmarkCommand(userCommand);
+            } catch (JarvisException e) {
+                DisplayMessages.outOfBounds();
+            }
+            break;
+
+        case "todo":
+            try {
+                todoCommand(userCommand);
+            } catch(JarvisException e) {
+                DisplayMessages.invalidInput();
+            }
+            break;
+
+        case "deadline":
+            try {
+                deadlineCommand(userCommand);
+            } catch(JarvisException e) {
+                DisplayMessages.invalidInput();
+            }
+            break;
+
+        case "event":
+            try {
+                eventCommand(userCommand);
+            } catch (JarvisException e) {
+                DisplayMessages.invalidInput();
+            }
+            break;
+        case "delete":
+            try {
+                deleteCommand(userCommand);
+            } catch (JarvisException e) {
+                DisplayMessages.invalidInput();
+            }
+            break;
```

Comment:
You might want to extract each of the `case` block into a separate method for SLAP. Alternatively, you can also combine everything into one `try-catch` block, or even handle the exceptions elsewhere.